docker/pravega/scripts/init_kubernetes.sh

```diff
@@ -42,29 +45,25 @@ init_kubernetes() {
 
         local ns=${POD_NAMESPACE}
         local podname=${POD_NAME}
+        export PUBLISHED_ADDRESS=""
+        export PUBLISHED_PORT=""
 
         service_type=$( k8 "${ns}" "services" "${podname}" ".spec.type" )
         if [ "${service_type}" == "LoadBalancer" ]; then
-            export PUBLISHED_ADDRESS=$( k8 "${ns}" "services" "${podname}" ".status.loadBalancer.ingress[0].ip" )
-            export PUBLISHED_PORT=$( k8 "${ns}" "services" "${podname}" ".spec.ports[].port" )
             while [ -z ${PUBLISHED_ADDRESS} ] || [ -z ${PUBLISHED_PORT} ]
             do
-                echo "LoadBalancer external address and port could not be obtained. Waiting 30 seconds and trying again..."
-                sleep 30
+                echo "Trying to obtain LoadBalancer external endpoint..."
+                sleep 10
                 export PUBLISHED_ADDRESS=$( k8 "${ns}" "services" "${podname}" ".status.loadBalancer.ingress[0].ip" )
```

Comment:
Might be a problem here. When `PUBLISHED_ADDRESS` is `null`, it can still exit the while loop. Maybe we can add something like `export PUBLISTSH_ADDRESS="null"` and `[ "${PUBLISHED_ADDRESS}" != "null" ]` . What do you think?

Reply:
@Tristan1900 with the changes made to the `k8` function, `PUBLISHED_ADDRESS` and `PUBLISHED_PORT` should never be "null". Am I missing something?

Reply:
Right! Sorry didn't see that.